### PR TITLE
add sdp_on_bf16 to tests,text-gen

### PR DESCRIPTION
**Pull request by hsubramony**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by jiminha** [comment]
@hsubramony  Do you need to update any Readme? , also can you update the Description?

File: tests/test_examples.py [hunk 1]
```diff
@@ -520,6 +520,26 @@ def test(self):
                 env_variables["PT_HPU_LAZY_MODE"] = "0"
                 env_variables["PT_ENABLE_INT64_SUPPORT"] = "1"
 
+            if self.EXAMPLE_NAME == "run_glue":
+                if model_name == "bert-large-uncased-whole-word-masking":
```
Comment (yeonsily) [inline tests/test_examples.py:524]: no need to add 'llama_guard' model? We have both in README.

File: tests/test_examples.py [hunk 2]
```diff
@@ -520,6 +520,26 @@ def test(self):
...
+                if model_name == "bert-large-uncased-whole-word-masking" or \
+                   model_name == "albert-large-v2":
+                    extra_command_line_arguments.append("--sdp_on_bf16")
+
+            if self.EXAMPLE_NAME == "run_bridgetower":
+                if model_name == "BridgeTower/bridgetower-large-itm-mlm-itc":
+                    extra_command_line_arguments.append("--sdp_on_bf16")
+
+            if self.EXAMPLE_NAME == "run_speech_recognition_seq2seq":
+                if model_name == "openai/whisper-small":
```
Comment (yeonsily) [inline tests/test_examples.py:537]: This is not updated in README. should I update?

File: examples/image-to-text/run_pipeline.py [hunk 3]
```diff
@@ -174,9 +174,15 @@ def main():
         action="store_true",
         help="Whether to use the key/value cache for decoding. It should speed up generation.",
     )
+    parser.add_argument(
```
Comment (yeonsily) [inline examples/image-to-text/run_pipeline.py:177]: --sdp_on_bf16 is not added to README. should I update all of the model in image-to-text also?

File: examples/text-generation/run_generation.py [hunk 4]
```diff
@@ -320,6 +320,9 @@ def setup_parser(parser):
         action="store_true",
         help="Run the inference with dataset for specified --n_iterations(default:5)",
     )
+    parser.add_argument(
```
Comment (yeonsily) [inline examples/text-generation/run_generation.py:323]: --sdp_on_bf16 is not added to README. should I update all of the model in run_generation also?

File: tests/test_fsdp_examples.py [hunk 5]
```diff
@@ -126,6 +127,7 @@ def _test_fsdp(
             "--use_flash_attention True",
             "--flash_attention_causal_mask True",
             f"--token {token.value}",
+            "--sdp_on_bf16",
```
Comment (yeonsily) [inline tests/test_fsdp_examples.py:130]: I see that this is common for both if/else case. Can we move it to the common command after line 83?

File: tests/test_examples.py (hunk 1 above)
Comment (hsubramony) [reply tests/test_examples.py:524]: which test ? i didnt see it ?

File: tests/test_examples.py (hunk 2 above)
Comment (hsubramony) [reply tests/test_examples.py:537]: we need to update readme for all these with PR 1514
contrastive-image-text/run_bridgetower.py
question-answering/run_qa.py
speech-recognition/run_speech_recognition_ctc.py
stable-diffusion//text_to_image_generation.py
text-classification/run_glue.py

File: examples/text-generation/run_generation.py (hunk 4 above)
Comment (hsubramony) [reply examples/text-generation/run_generation.py:323]: not sure for all models. thats a libin qtn

File: tests/test_examples.py (hunk 1 above)
Comment (yeonsily) [reply tests/test_examples.py:524]: https://github.com/huggingface/optimum-habana/blob/main/tests/utils.py#L79

File: tests/test_examples.py (hunk 2 above)
Comment (yeonsily) [reply tests/test_examples.py:537]: right. those READMEs are all updated. But I see that this model uses run_speech_recognition_seq2seq.py

File: examples/image-to-text/run_pipeline.py (hunk 3 above)
Comment (hsubramony) [reply examples/image-to-text/run_pipeline.py:177]: im only setting for "meta-llama/Llama-3.2-11B-Vision-Instruct"  and  "tiiuae/falcon-11B-vlm", so lets do only for those.

File: tests/test_examples.py (hunk 2 above)
Comment (hsubramony) [reply tests/test_examples.py:537]: yes we need update that too.

File: examples/image-to-text/run_pipeline.py (hunk 3 above)
Comment (yeonsily) [reply examples/image-to-text/run_pipeline.py:177]: this file change is already added by https://github.com/huggingface/optimum-habana/commit/89dfbb7b3ba1dd2a6a96e3f9a796d0d92bfdeea0

**Review by yeonsily — COMMENTED** [review]
LGTM

**Review by regisss — APPROVED** [review]

**Pull request merged**